app/Policies/InternetAccessPolicy.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Policies;
+
+use App\InternetAccess;
+use App\User;
+use App\Role;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class InternetAccessPolicy
+{
+    use HandlesAuthorization;
+
+    public function before($user, $ability)
+    {
+        if ($user->hasRole(Role::INTERNET_ADMIN)) {
+            return true;
+        }
+        if (!$user->hasRole(Role::INTERNET_USER)){
+            return false;
+        }
```

Comment:
Seems weird that this function conditionally returns `bool`.
What happens in case of *fallthrough*?

Reply:
The `before` method works as follows:
 * if it returns a result, it will be accounted for all gates.
 * if it doesn't, the queried gate will give the result